### PR TITLE
Make the inner button block not allowed as a reusable block or editable as HTML

### LINKED ISSUE #20942: Not able to convert reusable buttons child block to regular block.
I am going through some support issue on the WP repo support forum for Gutenberg.
I came across this support issue:
https://wordpress.org/support/topic/reusable-buttons/

**Describe the bug**
Clicking an individual button and saving it to a reusable block. One can not convert it back to a block again.

Adding a child/nested button to a reusable block.
Screen Shot 2020-03-16 at 22 52 34

Convert reusable nested button to regular block.
Screen Shot 2020-03-16 at 22 56 00

After clicking the Convert option nothing happens.

**To reproduce**
1. Add buttons block.
2. Click nested button. Then "Add to reusable blocks"
3. Notice there is no option to save. It automatically gets the title "Untitled Reusable Block"
4. Add the "Untitled Reusable Block" button to the layout.
5. Click to Convert it to regular block.
6. There is no reaction. The nested button reusable block is not converted.

**Expected behavior**
Not to be able to create a reusable block out of a child/nested buttons block. Here is a drop down menu example from selecting a Social icon child/nested block. Notice there is no option to add the nested icon to a reusable block.

Screen Shot 2020-03-16 at 23 01 36

**Desktop (please complete the following information):**
OSX
Gutenberg version 7.7.1
Desktop
Chrome

### The correct method to add a reusable buttons block:
1. Add some buttons. Or just have one.
2. Be sure that all the buttons are selected. It needs to look like this:
Screen Shot 2020-03-16 at 23 09 11

Or if only one button is in use:
Screen Shot 2020-03-16 at 23 10 39

You can now create a reusable block.

--------

If you have clicked the one button then it will look like this:
Screen Shot 2020-03-16 at 23 09 30

That means the single button is selected. Creating a reusable block when the single button is selected causes the error.

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #20942, which identifies that the button block can be made reusable, but then attempting to convert that reusable button block to a regular block outside of its parent context fails.

This PR makes the Button block consistent with other inner blocks (navigation link, social icons, column) with a distinct parent. Those blocks generally can't be made reusable, because of the above issue. They also generally can't be edited as HMTL. The general idea is that the parent has those options but the child doesn't.

With the Button block, there is a the possibility that this block is already in use in some reusable blocks and those still can't be converted to regular blocks, that's something this PR doesn't solve.

## How has this been tested?
1. Add a Buttons block
2. Try to make the Inner Button block reusable
3. Observe that there is now no menu option to achieve this.

## Screenshots 
Screenshot 2020-03-17 at 5 02 32 pm

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/button/index.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ export const settings = {
 	supports: {
 		align: true,
 		alignWide: false,
+		reusable: false,
+		html: false,
```
Comment (talldan) [inline packages/block-library/src/button/index.js]: It looks like this change affects mobile, so was wondering what the best way to update this setting on mobile would be (cc @pinarol)?

Is it best to keep the mobile setting as it currently is ... e.g.:

```js
export const settings = {
	...webSettings,
	supports: {
		align: true,
		alignWide: false,
	},
	parent: undefined,
};
```

**Comment by mtias** [comment]
I agree with the reusable bit, but not sure about the edit as HTML. Why shouldn't that be possible?

**Comment by talldan** [comment]
> I agree with the reusable bit, but not sure about the edit as HTML. Why shouldn't that be possible?

@mtias I noticed this was on other blocks with a parent, but it looks like I've made a bad assumption.

It's specified on the Column block because of that being a parent block (https://github.com/WordPress/gutenberg/pull/11785) and specified on the Navigation Link and Social Link blocks because those are dynamic blocks (https://github.com/WordPress/gutenberg/pull/19483).

So I've changed that again, the button can now be edited as html (74d3707).

**Comment by mtias** [comment]
Great, thanks, that makes more sense.

**Review by aduth — APPROVED** [review]

**Pull request merged**